PetGame/Controllers/LeaderboardController.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using PetGame.Core;
+using PetGame.Models;
+
+namespace PetGame
+{
+    [Route("api/[controller]")]
+    public class LeaderboardController : Controller
+    {
+        private readonly SqlManager sqlManager;
+        private readonly LeaderboardService LeaderboardService;
+
+        public LeaderboardController(SqlManager sqlManager)
+        {
+            this.sqlManager = sqlManager;
+            LeaderboardService = new LeaderboardService(sqlManager);
+        }
+
+        /// <summary>
+        /// Returns a List of LeaderboardEntry as JSON.
+        /// Can be offset and the number of results can be changed
+        /// using a raw JSON request body
+        /// </summary>
+        /// <param name="Request"> 
+        /// JSON request body containing offset and desired number of results
+        /// Defaults to 0 offset and 10 results if no request body is present
+        /// </param>
+        /// <returns>
+        /// Each LeaderboardEntry contains: Score, Timestamp, PetName, PetId
+        /// OwnerName, OwnerId
+        /// </returns>
+        [HttpGet]
+        public IActionResult Get([FromBody] LeaderboardRequestModel Request)
```

Comment:
This model can never be supplied with a `GET`, I'd add a `[HttpPost("options")]` route that allows for these options to be specified in the json body.

Reply:
Add, or replace?

Reply:
Add